pkg/platform/runtime/setup/buildlog/artifactlog.go

```diff
@@ -0,0 +1,55 @@
+package buildlog
+
+import (
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+)
+
+// ArtifactLog wraps a handler to listen for real-time build log messages for a specific artifact
+type ArtifactLog struct {
+	errCh chan error
+}
+
+// NewArtifactLog subscribes to events on the connection, and forwards build log events via the events handler
+func NewArtifactLog(artifactID artifact.ArtifactID, conn BuildLogConnector, events Events) (*ArtifactLog, error) {
+	errCh := make(chan error)
+
+	go func() {
+		defer close(errCh)
+
+		for {
```

Comment:
This seems like it may need something more to ensure it comes to an end. Context type, maybe. I also wonder if the error returned by conn.ReadJSON could be EOF and whether that can be handled here for callers such that the loop ends, but no error is returned.

Reply:
Unfortunately, the `conn.ReadJSON` method does not support context timeouts.  So, I decided to just always exit without an error if `ReadJSON` returns.
It is also unfortunate, that the logs do not have a clear end-of-stream marker, because I would feel much more comfortable returning from the for-loop on an actual message that we receive.